### ISSUE TITLE
Very slow build on worker

### ISSUE THREAD

**Issue by kgrach** [fastbuild/fastbuild#416]
I use distributed compilation feature and to see very slow compilation on worker machine.

For example, same file NDFLRequestProcessingEx has compile duration time
1) local 0:22
2) worker over 5:55

This problem present always when any file compile on a worker machine.
What could be the problem? 
Local and distibuted machines are identical core speed, disk i/o, network i/o. Network speed is 10Gb.

![slowworkerbuild](https://user-images.githubusercontent.com/9927954/38912444-7ce02da8-42ff-11e8-8069-e0eae4d552fb.jpg)

**Comment by ffulin** [comment]
It is normal for remote compile times to be longer on remote machines, but certainly not to this degree.

One thing that springs to mind for you to check is the impact of any Precompiled Headers you may be using. Precompiled Headers are not used on the remote machine. In effect, the code in the PCH is also compiled as part of each object when objects are compiled remotely. For "normal" PCH use this overhead on the remote compilation is greatly outweighed by the addition of the remote cores. If your PrecompiledHeader is very large however, this could be a factor (if your PCH takes several minutes to build for example).

Other than that, I think you'd have to try to gain some more information about what is going on on the remote machine by profiling it - i.e. seeing if it is CPU or disk bound during that 5 min period, or if it is waiting on a network transfer (either incoming or outgoing)?

Finally, assuming you are using the MSVC compiler, some other stabs in the dark:
 - Are you using the Windows runtime (/ZW) option? If so, does the problem occur for all objects, or just those using this option.
 - Are you using any other "exotic" compiler options? As above.
 - Are you using /Z7 of /Zi debug format? Try switching between the two if you can.

Let me know if that helps.

**Comment by ffulin** [comment]
@kgrach Did you make any progress on this issue?

**Comment by kgrach** [comment]
Sorry, no progress.

**Comment by ffulin** [comment]
I'm going to close this issue for now. Let me know if you want to continue troubleshooting and we can re-open if needed, or raise a new issue depending on what your next investigation yields.
